### DISCUSSION TITLE
Deploy with policies.json and define Settings

### DISCUSSION THREAD

**Discussion by critkitten** [tridactyl/tridactyl#4515] [Q&A] [answered]
With ublock it's possible to define some options and deploy system wide with:
You setup ublock and export settings and copy the text file content into http://raymondhill.net/ublock/adminSetting.html to convert it into a valid string.
```
    "3rdparty": {
      "Extensions": {
        "uBlock0@raymondhill.net": {
          "adminSettings": {
             "timeStamp":X,"version":"X","userSettings":{"advancedUserEnabled":true,"importedLists":[],"popupPanelSections":31},"selectedFilterLists":["user-filters","ublock->
          }
        }
      }
    },
```

Is this possible with tridactyl? I ask because i deploy tridactyl with
```
      "tridactyl.vim@cmcaine.co.uk": {
        "install_url": "https://addons.mozilla.org/firefox/downloads/file/4036604/tridactyl_vim-1.23.0.xpi",
        "installation_mode": "force_installed"
      },
```
but every time i remove my browser profile i start at beginning with settings reset and annoying startup page i don't need.

**Comment by bovine3dom** [comment, marked as answer]
It looks like uBlock is using a special API to read from policies.json: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/managed

via

https://github.com/gorhill/uBlock/blob/fa5e4b7769c9759a588a4b842d79d3273c5756df/platform/common/vapi-background.js#L1323

And

https://github.com/gorhill/uBlock/blob/a77be2a17230a02692c4ab35a22156dade3bcdde/src/js/storage.js#L1312

We would need to add more code to Tridactyl to support that.

**Reply by critkitten** [reply]
thanks for the explanation.

**Comment by m040601** [comment]
> define some options and deploy system wide with

Very, very interesting stuff indeed.

> It looks like uBlock is using a special API to read from policies.json: 

I wasnt even aware this was possible with ublock origin. Or the mozilla "policies" and "manifestos" deployment thing. Really powerfull stuff. Really worth spending time with it. So much work to do just to safely grab a few bytes from some other place in the world with the "modern" web :-)

Thanks for critkitten, for bringing this up, and bovine3dom for further insights.


On Arch Linux you can get ublock and tridactyl as official packages. No need to setup addons for each new/fresh user. Saves you a trip to addons.mozilla.org. But you still have to do the dance of exporting old settings and reimporting them again.



Somehow had never ocurred to me to check out their wiki, for this kind  (and others) of advanced/enterprisy  setup. A must read.for those of us for whom Firefox without ublock and tridactyl (and a hardened user.js) simply doesnt exist.

This automated deployment in freshly installed new systems, or throwaway profiles is a dream. Having not only the software, but also all your prior stable customizations and decisions some how scripted, firing up the browser and not needing to go through the setup again is a huge time saver.

> We would need to add more code to Tridactyl to support that.

Hope there might be some progress in the future for that.
